addons/Drop/drop.lua

```diff
@@ -0,0 +1,65 @@
+_addon.name = 'Drop'
+_addon.author = 'Arico'
+_addon.version = '1'
+_addon.commands = {'drop'}
+
+
+packets = require('packets')
+require('luau')
+res = require('resources')
+require('logger')
+
+check_inv = function(item_id)
+        for k, v in pairs(windower.ffxi.get_items().inventory) do
+            if type(v) == 'table' and v.id == item_id then
+                return true
+            end
+        end
+    return false
+end
+get_item_resource = function(item)

```

Comment:
This is a very slow function, and unnecessary. *If* you decide to keep it, call `item:lower()` only once at the start and use the resulting variable. The way it is now you create a lowercase string of it up to 44 thousand times. Also, Japanese does not have casing, and calling `v.japanese:lower()` could actually mess up the resulting string.

But like I said, this entire function is unnecessary. You don't need to iterate all the resources, with its 22 thousand items. Instead it's enough to iterate the inventory of max 80 items, get the item's ID and use the resources to look up by ID, which is fast. Then you can compare the `english` and `japanese` of the found item (if it exists) with the user's string. You can even enable wildcard matching like this, so the user can do something like `//drop * crystal` and have it drop all crystals.